Source/MBF.SAMCD.ADC.pas

```diff
@@ -0,0 +1,225 @@
+unit MBF.SAMCD.ADC;
+{
+  This file is part of Pascal Microcontroller Board Framework (MBF)
+  Copyright (c) 2015 -  Michael Ring
+  based on Pascal eXtended Library (PXL)
+  Copyright (c) 2000 - 2015  Yuriy Kotsarenko
+
+  This program is free software: you can redistribute it and/or modify it under the terms of the FPC modified GNU
+  Library General Public License for more
+
+  This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the FPC modified GNU Library General Public
+  License for more details.
+}
+{< Atmel SAMD series ADC functions. }
+
+interface
+
+{$include MBF.Config.inc}
+
+uses
+  MBF.SAMCD.Helpers,
+  MBF.SAMCD.GPIO;
+
+const
+{$if defined(samd10) or defined(samd21)}
+  ADCMux = PORT_PMUX_B_Val;
+{$elseif defined(samc21)}
+  ADCMux = PORT_PMUX_B_Val;
+  ADC1Mux = PORT_PMUX_B_Val;
+{$else}
+  {$error Unknown Chip, please check mbf.boards.samd10.inc and then define ADC Mux here}
+{$endif}
+
+type
+  TADCHelper = record helper for TADC_Registers
+  private
+    procedure SyncWait;
+    procedure ConversionWait;
+    function GetRawValue(const Channel: TPinIdentifier): Cardinal;
+  public
+    procedure Initialize;
+    property GetADCResult[const Channel: TPinIdentifier]:Cardinal read GetRawValue;
+  end;
+
+implementation
+
+uses
+  MBF.BitHelpers,
+  MBF.SAMCD.SystemCore;
+
+type
+  {$ifdef samd10}
+  TADCPin = (AIN0,AIN1,AIN2,AIN3,AIN4,AIN5,AIN6,AIN7,AIN8,AIN9,None);
+  {$endif}
+  {$ifdef samd21}
+  TADCPin = (AIN0,AIN1,AIN2,AIN3,AIN4,AIN5,AIN6,AIN7,AIN8,AIN9,AIN10,AIN11,AIN12,AIN13,AIN14,AIN15,AIN16,AIN17,AIN18,AIN19,None);
+  {$endif}
+  {$ifdef samc21}
+  TADCPin = (AIN0,AIN1,AIN2,AIN3,AIN4,AIN5,AIN6,AIN7,AIN8,AIN9,AIN10,AIN11,None);
+  TADC1Pin = (AIN0,AIN1,AIN2,AIN3,AIN4,AIN5,AIN6,AIN7,AIN8,AIN9,AIN10,AIN11,None);
+  {$endif}
+
+  TADCPins = set of TADCPin;
+  {$ifdef samc21}
+  TADC1Pins = set of TADC1Pin;
+  {$endif}
+
+const
+  {$ifdef samd10}
+  TADCPinsMap : array[TADCPin] of TPinIdentifier=(
```

Comment:
Please try to use definitions as done in my SPI code (For example mbf.stm32f4.spi TSPIMOSIPins
The advantage is that you get a choice of all available analog pins in Lazarus which makes coding a lot easier.

Reply:
Will look into it !